src/main/python/ayab/engine/control.py

```diff
@@ -226,11 +262,12 @@ def select_needles_API6(self, color, row_index, blank_line):
 
         return bits
 
-    def operate(self, operation):
+    def operate(self, operation:Operation)->Output:
         """Finite State Machine governing serial communication"""
         method = "_API" + str(self.api_version) + "_" + self.state.name.lower()
         if not hasattr(StateMachine, method):
+            #TODO: yell about this maybe?
             return Output.NONE
-        dispatch = getattr(StateMachine, method)
+        dispatch:Callable[[Control,Operation],Output] = getattr(StateMachine, method)
         result = dispatch(self, operation)
         return result
```

Comment:
I like this one because the type of `result` is properly inferred as `Output` purely from the `Callable` type there.